t/plugin/openid-connect7.t

```diff
@@ -0,0 +1,426 @@
+                                "use_jwks": true,
+                                "realm": "University"
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 8: Obtain valid token and access route with it. Use valid_issuer from discovery endpoint.
+--- http_config
+    server {
+        listen 8089;
+
+        location /realms/University/.well-known/openid-configuration {
+            content_by_lua_block {
+                ngx.say([[
+{
+  "issuer": "http://127.0.0.1:8089/realms/University",
+  "jwks_uri": "http://127.0.0.1:8089/realms/University/protocol/openid-connect/certs",
+  "response_types_supported": ["id_token"],
+  "subject_types_supported": ["public"],
+  "id_token_signing_alg_values_supported": ["RS256"]
+}]])
+            }
+        }
+    }
+--- config
+    location /t {
+        content_by_lua_block {
+            -- Obtain valid access token from Keycloak using known username and password.
+            local json_decode = require("toolkit.json").decode
+            local http = require "resty.http"
+            local httpc = http.new()
+            local uri = "http://127.0.0.1:8080/realms/University/protocol/openid-connect/token"
+            local res, err = httpc:request_uri(uri, {
+                    method = "POST",
+                    body = "grant_type=password&client_id=course_management&client_secret=d1ec69e9-55d2-4109-a3ea-befa071579d5&username=teacher@gmail.com&password=123456",
+                    headers = {
+                        ["Content-Type"] = "application/x-www-form-urlencoded"
+                    }
+                })
+
+            -- Check response from keycloak and fail quickly if there's no response.
+            if not res then
+                ngx.say(err)
+                return
+            end
+
+
+            -- Get access token from JSON response body.
+            local body = json_decode(res.body)
+            local accessToken = body["access_token"]
+
+            -- Access route using access token. Should work.
+            uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/hello"
+            local res, err = httpc:request_uri(uri, {
+                method = "GET",
+                headers = {
+                    ["Authorization"] = "Bearer " .. body["access_token"]
+                }
+                })
+        }
+    }
+--- error_log
+valid_issuers not provided, using issuer from discovery doc: http://127.0.0.1:8089/realms/University
```

Comment:
No token verify succeeded assertion

Reply:
Actually here I am mocking in order to override the discovery endpoint output. But the token generation takes place in real server so that token cant be used to verify with mock discovery server

Reply:
In above tests I have proved the working of valid_issuers field and here I am asserting that the overridden issuer was assigned to `valid_issuers`.

Reply:
I still don't quite understand why `TEST 6: Obtain valid token and access route with it.` can verify the token, but this test case can't.

Reply:
modified